core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -337,6 +338,8 @@ object KafkaConfig {
   val SslClientAuthProp = SslConfigs.SSL_CLIENT_AUTH_CONFIG
 
   /** ********* SASL Configuration ****************/
+  val SaslMechanismProp = SaslConfigs.SASL_MECHANISM
```

Comment:
It seems that this is used for inter-broker communication. Shouldn't we be using the same pattern we used for `InterBrokerSecurityProtocol`?